EIP-1: Mark sections that are optional as optional (closes #5112) (#5118)
ethereum/EIPs: EIPS/eip-1.md

- Preamble - RFC 822 style headers containing metadata about the EIP, including the EIP number, a short descriptive title (limited to a maximum of 44 characters), a description (limited to a maximum of 140 characters), and the author details. Irrespective of the category, the title and description should not include EIP number. See [below](./eip-1.md#eip-header-preamble) for details.
- Abstract - Abstract is a multi-sentence (short paragraph) technical summary. This should be a very terse and human-readable version of the specification section. Someone should be able to read only the abstract to get the gist of what this specification does.

> - Motivation _(optional)_ - A motivation section is critical for EIPs that want to change the Ethereum protocol. It should clearly explain why the existing protocol specification is inadequate to address the problem that the EIP solves. This section may be omitted if the motivation is extremely obvious, but EIP submissions without sufficient motivation may be rejected outright.

Reviewer: I'm not a fan of applying editorial judgement on the optionality here.  Can we remove that last bit?
```suggestion
- Motivation _(optional)_ - A motivation section is critical for EIPs that want to change the Ethereum protocol. It should clearly explain why the existing protocol specification is inadequate to address the problem that the EIP solves. This section may be omitted if the motivation is extremely obvious.
```

- Specification - The technical specification should describe the syntax and semantics of any new feature. The specification should be detailed enough to allow competing, interoperable implementations for any of the current Ethereum platforms (cpp-ethereum, go-ethereum, parity, ethereumJ, ethereumjs-lib, [and others](https://github.com/ethereum/wiki/wiki/Clients).
- Rationale - The rationale fleshes out the specification by describing what motivated the design and why particular design decisions were made. It should describe alternate designs that were considered and related work, e.g. how the feature is supported in other languages. The rationale should discuss important objections or concerns raised during discussion around the EIP.

> - Backwards Compatibility _(optional)_ - All EIPs that introduce backwards incompatibilities must include a section describing these incompatibilities and their severity. The EIP must explain how the author proposes to deal with these incompatibilities. This section may be omitted if the proposal does not introduce any backwards incompatibilities, but EIP submissions without a sufficient backwards compatibility section may be rejected outright.

Reviewer: I'm surprised by this one.  I always thought Backwards Compatibility was required.  As above, I would like to remove the editor subjectivity:
```suggestion
- Backwards Compatibility _(optional)_ - All EIPs that introduce backwards incompatibilities must include a section describing these incompatibilities and their severity. The EIP must explain how the author proposes to deal with these incompatibilities. This section may be omitted if the proposal does not introduce any backwards incompatibilities, but this section must be included if backward incompatibilities exist.
```

> - Backwards Compatibility _(optional)_ - All EIPs that introduce backwards incompatibilities must include a section describing these incompatibilities and their severity. The EIP must explain how the author proposes to deal with these incompatibilities. This section may be omitted if the proposal does not introduce any backwards incompatibilities, but EIP submissions without a sufficient backwards compatibility section may be rejected outright.

Reviewer 2: I'd say that consequences are better because leaving it as "severity" would mean that something like this would be allowed (this is an example, it's meaningless):

`functionName` requires the actual value of `variable` - low severity

Whereas with consequences:

`functionName` takes `hash(variable)` as an argument: this means that two mappings will need to be stored, one indexed by `hash(variable)` and one indexed by `variable`.

This would help people implementing that EIP, because it avoids mistakes like implementing the EIP with, say, only one of the mappings until they try to implement `functionName`.

> - Test Cases _(optional)_ - Test cases for an implementation are mandatory for EIPs that are affecting consensus changes. Tests should either be inlined in the EIP as data (such as input/expected output pairs, or included in `../assets/eip-###/<filename>`. This section may be omitted for non-Core proposals, but EIP submissions without sufficient test cases may be rejected outright.

Reviewer: ```suggestion
- Test Cases _(optional)_ - Test cases for an implementation are mandatory for EIPs that are affecting consensus changes. Tests should either be inlined in the EIP as data (such as input/expected output pairs, or included in `../assets/eip-###/<filename>`. This section may be omitted for non-Core proposals.
```

> - Test Cases _(optional)_ - Test cases for an implementation are mandatory for EIPs that are affecting consensus changes. Tests should either be inlined in the EIP as data (such as input/expected output pairs, or included in `../assets/eip-###/<filename>`. This section may be omitted for non-Core proposals, but EIP submissions without sufficient test cases may be rejected outright.

Reviewer: While I agree that test cases are generally a good idea, I'm not a huge fan of adding that as a hard requirement as it raises the barrier of entry to EIP authorship.  I also worry that it will result in *even more* EIPs that get stuck in Draft/Review forever (and eventually moved to stagnant) while they are actively used within the community.
